### PR TITLE
MBS-10781: Add support for Traxsource URLs

### PR DESCRIPTION
**Pull request by atj**
This commit implements cleaning, validation and normalisation of
Traxsource URLs when added to artists, labels, releases and recordings.

### CODE REVIEW

**Comment by atj** [comment]
Would you like me to create a ticket in the issue tracker for this?

**Comment by yvanzo** [comment]
@brainzbot, test this please.

File: root/static/scripts/tests/Control/URLCleanup.js [hunk 1]
```diff
@@ -3195,6 +3195,56 @@ const testData = [
     expected_relationship_type: 'patronage',
             expected_clean_url: 'https://www.tipeee.com/example',
   },
+  // Traxsource
+  {
+                     input_url: 'https://www.traxsource.com/artist/584/joey-negro
```
Comment (yvanzo) [inline root/static/scripts/tests/Control/URLCleanup.js:3200]:
```suggestion
                     input_url: 'https://www.traxsource.com/artist/584/joey-negro',
```

Comment (atj) [reply root/static/scripts/tests/Control/URLCleanup.js]: Thanks, this is fixed now.

**Comment by atj** [comment]
I've just fixed another syntax issue which caused the tests to fail.

Is there a guide anywhere on how to run the test suite when developing using the Docker instances?

**Comment by atj** [comment]
> Hi @atj, thank you for submitting this patch. Yes, having a ticket for that change would be ideal. Apart a pair of missing characters, the rest of the PR looks perfect to me.

I've created ticket [MBS-10781](https://tickets.metabrainz.org/browse/MBS-10781) for this.

**Comment by yvanzo** [comment]
> Is there a guide anywhere on how to run the test suite when developing using the Docker instances?

For a development setup using Docker Compose, see https://github.com/metabrainz/musicbrainz-docker/tree/mbvm-38-dev#development-setup

To run JavaScript tests in your browser:

1. Log into the Docker instance of musicbrainz:
   ```bash
   sudo docker-compose exec musicbrainz bash
   ```
2. Compile tests
   ```bash
    script/compile_resources.sh tests
    ```
3. Go to http://localhost:5000/static/scripts/tests/web.html

This is inspired by https://github.com/metabrainz/musicbrainz-server/blob/master/HACKING.md#javascript which is not yet fully supported by Docker development setup.

**Pull request merged**